src/otx/algo/classification/backbones/vision_transformer.py

```diff
@@ -5,612 +5,485 @@
+        self.norm_pre = norm_layer(embed_dim) if pre_norm else nn.Identity()
+
+        dpr = [x.item() for x in torch.linspace(0, drop_path_rate, depth)]  # stochastic depth decay rule
+        self.blocks = nn.Sequential(
+            *[
+                block_fn(
+                    dim=embed_dim,
+                    num_heads=num_heads,
+                    mlp_ratio=mlp_ratio,
+                    qkv_bias=qkv_bias,
+                    qk_norm=qk_norm,
+                    init_values=init_values,
+                    proj_drop=proj_drop_rate,
+                    attn_drop=attn_drop_rate,
+                    drop_path=dpr[i],
+                    norm_layer=norm_layer,
+                    act_layer=act_layer,
+                    mlp_layer=mlp_layer,
+                )
+                for i in range(depth)
+            ],
+        )
 
-        Returns:
-            nn.Module: The normalization layer.
-        """
-        return self.ln2
+        self.norm = norm_layer(embed_dim)
 
     def init_weights(self) -> None:
         """Initializes the weights of the VisionTransformer."""
         super().init_weights()
 
-        if not (
-            isinstance(self.init_cfg, dict) and self.init_cfg["type"] == "Pretrained"
-        ) and self.pos_embed is not None:
+        if self.pos_embed is not None:
             trunc_normal_(self.pos_embed, std=0.02)
 
-    def _prepare_pos_embed(self, state_dict: dict, prefix: str, *args, **kwargs) -> None:
-        name = prefix + "pos_embed"
-        if name not in state_dict:
-            return
-
-        ckpt_pos_embed_shape = state_dict[name].shape
-        if not self.with_cls_token and ckpt_pos_embed_shape[1] == self.pos_embed.shape[1] + 1:
-            # Remove cls token from state dict if it's not used.
-            state_dict[name] = state_dict[name][:, 1:]
-            ckpt_pos_embed_shape = state_dict[name].shape
-
-        if self.pos_embed.shape != ckpt_pos_embed_shape:
-            embed_size = int(np.sqrt(ckpt_pos_embed_shape[1] - self.num_extra_tokens))
-            # ckpt_pos_embed_shape = to_2tuple(
-            #     int(np.sqrt(ckpt_pos_embed_shape[1] - self.num_extra_tokens)))
-            ckpt_pos_embed_shape = (embed_size, embed_size)
-            pos_embed_shape = self.patch_embed.init_out_size
-
-            state_dict[name] = resize_pos_embed(
-                state_dict[name],
-                ckpt_pos_embed_shape,
-                pos_embed_shape,
-                self.interpolate_mode,
-                self.num_extra_tokens,
-            )
-
-    @staticmethod
-    def resize_pos_embed(*args, **kwargs) -> torch.Tensor:
-        """Interface for backward-compatibility."""
-        return resize_pos_embed(*args, **kwargs)
+    @torch.jit.ignore()
+    def load_pretrained(self, checkpoint_path: str, prefix: str = "") -> None:
+        """Loads the pretrained weight to the VisionTransformer."""
+        checkpoint_ext = checkpoint_path.split(".")[-1]
```

Comment:
```suggestion
    def load_pretrained(self, checkpoint_path: Path, prefix: str = "") -> None:
        """Loads the pretrained weight to the VisionTransformer."""
        checkpoint_ext = checkpoint_path.suffix()
```

Reply:
Could you explain why this is better?

Reply:
`cache_file` in the below is already `pathlib.Path` and we can use its functionality as is without converting it to `str`.
And I think it would be better to use path-related tools (`os.path` or `pathlib.Path`) than string-related tools.

Reply:
fixed at 2fe16784c73b37ac23a3c7d9477afeab6f886a10